### PR TITLE
Adds sample painless script to extract n-grams

### PR DESCRIPTION
**Pull request by Winterflower**

### CODE REVIEW

File: Machine Learning/DGA Detection/README.md [hunk 1]
```diff
@@ -1,2 +1,111 @@
...
+We are interested in extracting unigrams, bigrams and trigrams from the 
+`domain` field. Thus, we have to first define a Painless script that is
+capable of taking a string as an input and expanding the string
+into a set of n-grams, for some value of `n`. 
+
+The script is available in the file `ngram-extractor-reindex.json`, but is also
+reproduced below. 
+
+You can store it in Elasticsearch using the following Dev Console command.
+As you can see, the scripts accepts one parameters with is `n`, the length
```
Comment (joshdevins) [inline Machine Learning/DGA Detection/README.md:27]: "the script accepts one parameter which is" (singular) (spelling)

File: Machine Learning/DGA Detection/README.md [hunk 2]
```diff
@@ -1,2 +1,111 @@
...
+for (int i=0;i<ctx['domain'].length();i++){
+  ctx[Integer.toString(params.ngram_count)+'-gram_field'+Integer.toString(i)] = nGramAtPosition(ctx['domain'], i, params.ngram_count)
+}
+ """
+  }
+}
+
+```
+
+We can then use the stored scripts to configure an Ingest Pipeline as follows.
```
Comment (joshdevins) [inline Machine Learning/DGA Detection/README.md:56]: "script" (singular)

File: Machine Learning/DGA Detection/README.md [hunk 3]
```diff
@@ -1,2 +1,111 @@
...
+
+
+```
+PUT _ingest/pipeline/dga_ngram_expansion_reindex
+{
+    "description": "Expands a domain into chars, bigrams and trigrams",
```
Comment (joshdevins) [inline Machine Learning/DGA Detection/README.md:62]: "... unigrams, bigrams and trigrams"

File: Machine Learning/DGA Detection/README.md [hunk 4]
```diff
@@ -1,2 +1,111 @@
...
+            "ngram_count":3
+          }
+        }
+      }
+    ]
+}
+```
+
+Once the Ingest Pipeline has been configured we can re-index
+our original index with the raw data into a new index which contains the
```
Comment (joshdevins) [inline Machine Learning/DGA Detection/README.md:93]: "... which will contain" ?

**Comment by Winterflower** [comment]
Thank you for the review @joshdevins !
Have modified according to the changes you requested.

**Pull request merged**